packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1688 @@
+    this.cookie,
+    this.setCookie,
+    CustomHeaders? custom,
+
+    // Security and Modern Headers
+    this.strictTransportSecurity,
+    this.contentSecurityPolicy,
+    this.referrerPolicy,
+    this.permissionsPolicy,
+    this.accessControlAllowMethods,
+    this.accessControlAllowHeaders,
+    this.clearSiteData,
+    this.secFetchDest,
+    this.secFetchMode,
+    this.secFetchSite,
+    this.crossOriginResourcePolicy,
+    this.crossOriginEmbedderPolicy,
+    this.crossOriginOpenerPolicy,
+    required this.failedHeadersToParse,
+  }) : custom = custom ?? CustomHeaders.empty();
+
+  factory Headers.fromHttpRequest(
+    io.HttpRequest request, {
+    bool strict = false,
+    required String? xPoweredBy,
+  }) {
+    Map<String, List<String>> failedHeadersToParse = {};
+    var dartIOHeaders = request.getHeadersParser(
+      strict: strict,
+      onHeaderFailedToParse: (String key, List<String> value) {
+        // We dont remove empty values because we want to save the
+        // original value as it is, so we can see what was the invalid value
+        value = value.splitTrimAndFilterUnique(
+          emptyCheck: false,
+        );
+        if (failedHeadersToParse.containsKey(key)) {
+          failedHeadersToParse[key]!.addAll(value);
+        } else {
+          failedHeadersToParse[key] = value;
+        }
+      },
+    );
+
+    return _HeadersImpl(
+      // Date-related headers
+      date: dartIOHeaders.parseDate(_dateHeader),
+      expires: dartIOHeaders.parseDate(_expiresHeader),
+      ifModifiedSince: dartIOHeaders.parseDate(_ifModifiedSinceHeader),
+      lastModified: dartIOHeaders.parseDate(_lastModifiedHeader),
+
+      // General Headers
+      origin: dartIOHeaders.parseUri(_originHeader),
+      server: dartIOHeaders.parseString(
+        _serverHeader,
+      ),
+      via: dartIOHeaders.parseStringList(
+        _viaHeader,
+      ),
+
+      // Request Headers
+      from: dartIOHeaders.parseMultipleValue(
+        _fromHeader,
+        onParse: FromHeader.parse,
+      ),
+      host: dartIOHeaders.parseUri(_hostHeader),
+      acceptEncoding: dartIOHeaders.parseMultipleValue(
+        _acceptEncodingHeader,
+        onParse: AcceptEncodingHeader.parse,
+      ),
+      acceptLanguage: dartIOHeaders.parseMultipleValue(
+        _acceptLanguageHeader,
+        onParse: AcceptLanguageHeader.parse,
+      ),
+      accessControlRequestHeaders: dartIOHeaders.parseStringList(
+        _accessControlRequestHeadersHeader,
+      ),
+      accessControlRequestMethod: dartIOHeaders.parseSingleValue(
+        _accessControlRequestMethodHeader,
+        onParse: Method.parse,
+      ),
+      age: dartIOHeaders.parseInt(
+        _ageHeader,
+        allowNegative: false,
+      ),
+      allow: dartIOHeaders.parseMultipleValue(
+        _allowHeader,
+        onParse: (values) {
+          return values
+              .splitTrimAndFilterUnique(emptyCheck: false)
+              .map(Method.parse)
```

Comment:
Suggestion: Follow the general pattern and avoid implementation in this declaration.